### PR TITLE
 docs: translated to ES-LA exercises 3,4,5 and 6

### PR DESCRIPTION
**Pull request by ErickRodrCodes**
Added to ES-LA translations for Angular Exercises 3,4,5 and 6.

## Checklist for challenge submission

- [x] Start your PR message with Answer:${challenge_number}

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,7 +19,6 @@ node_modules
 
 # IDE - VSCode
 .vscode/*
-!.vscode/settings.json
```
Comment (tomalaforge) [inline .gitignore:22]: why did you delete this one ?

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-  "eslint.validate": ["json"]
+  "eslint.validate": [
+    "json"
+  ],
+  "cSpell.language": "en,es-ES"
```
Comment (tomalaforge) [inline .vscode/settings.json:5]: I prefer to keep this file ignore for now. Everyone has its own extension.

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+description: El desafío 3 se trata de una directive incorporada.
+author: thomas-laforge
+challengeNumber: 3
+command: angular-ngfor-enhancement
+blogLink: https://medium.com/@thomas.laforge/ngfor-enhancement-716b44656a6c
+sidebar:
+  order: 101
+---
+
+:::nota
```
Comment (tomalaforge) [inline docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:12]: this is a markdown keywork. Should stay :::note

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+Las Directivas en Angular Framework son una herramienta muy poderosa. Usted puede aplicar los principios de DRY para aplicar una lógica compartida dentro de una directiva y aplicarla a cualquier componente.
+
+Pero el verdadero poder esta en mejorar una directiva ya existente que además **no te pertenece**.
+
+## Declaración
+
+En este desafío, queremos mostrar una lista de personas. Si la lista esta vacía, usted deberá mostrar _" the list is empty !! "_. (La lista esta vacía)
+
+:::nota del traductor
```
Comment (tomalaforge) [inline docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:26]: same here

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+challengeNumber: 4
+command: angular-context-outlet-type
+blogLink: https://medium.com/@thomas.laforge/ngtemplateoutlet-type-checking-5d2dcb07a2c6
+sidebar:
+  order: 201
+---
+
+## Información
+
+:::Nota del traductor
```
Comment (tomalaforge) [inline docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md:14]: same

File: docs/src/content/docs/es/challenges/angular/6-permissions.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+Las Directivas Estructurales son in concepto importante que necesitaras dominar para mejorar tus habilidades y conocimientos en Angular. Esta es la primera parte del desafío.
+
+En Angular, "Guard" es un concepto muy importante por que siempre las vas a necesitar en cada aplicación que construyas.
+
+## Declaración
+
+En LoginComponent, encontraras 6 botones correspondientes a 6 tipos de roles de usuario:
+
+:::Notas del traductor
```
Comment (tomalaforge) [inline docs/src/content/docs/es/challenges/angular/6-permissions.md:22]: same

**Comment by tomalaforge** [comment]
Thanks a lot for your work 👍, you need to check the checkbox inside the ES issue.

File: .gitignore (hunk 1 above)
Comment (ErickRodrCodes) [reply .gitignore:22]: ok, when using code spell checker, the settings of the current user languages are being set here. so to prevent conflicts with other translators, I skipped this file.

**Comment by ErickRodrCodes** [comment]
NP, I will rebase with the needed changes.

File: .vscode/settings.json (hunk 2 above)
Comment (ErickRodrCodes) [reply .vscode/settings.json:5]: it is referenced on the gitIgnore as

`.vscode/*`

and by removing 

`!.vscode/settings.json`

from it.

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+## Declaración
+
+En este desafío, queremos mostrar una lista de personas. Si la lista esta vacía, usted deberá mostrar _" the list is empty !! "_. (La lista esta vacía)
+
+:::note
+_(Notas del traductor)_
+
+En caso que hayan test automáticos para validar este desafío, he preferido dejar la frase en ingles, en vez de utilizar una traducción directa al español.
+:::
```
Comment (tomalaforge) [inline docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:30]: Why did you added this ?

Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:30]: ok, sounds fair. I can edit it to reflect it in a more generic scenario rather than a note from the translator.

File: docs/src/content/docs/es/challenges/angular/5-crud.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+title: 🟢 Aplicación CRUD
+description: El desafío 5 se trata de refactorizar una aplicación CRUD.
+author: thomas-laforge
+challengeNumber: 5
+command: angular-crud
+sidebar:
+  order: 2
+---
+
+## Information
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/5-crud.md:11]: Looks like we missed this title, can we add a translation here as well?

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+description: El desafío 3 se trata de una directive incorporada.
+author: thomas-laforge
+challengeNumber: 3
+command: angular-ngfor-enhancement
+blogLink: https://medium.com/@thomas.laforge/ngfor-enhancement-716b44656a6c
+sidebar:
+  order: 101
+---
+
+:::note
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:12]: Nitpick: this `Note` title I believe can also be translated. I believe the syntax is:

```
:::note[Nota]

text

:::
```

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+Las Directivas en Angular Framework son una herramienta muy poderosa. Usted puede aplicar los principios de DRY para aplicar una lógica compartida dentro de una directiva y aplicarla a cualquier componente.
+
+Pero el verdadero poder esta en mejorar una directiva ya existente que además **no te pertenece**.
+
+## Declaración
+
+En este desafío, queremos mostrar una lista de personas. Si la lista esta vacía, usted deberá mostrar _" the list is empty !! "_. (La lista esta vacía)
+
+:::note
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:26]: same comment regarding `Note` translation

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+challengeNumber: 4
+command: angular-context-outlet-type
+blogLink: https://medium.com/@thomas.laforge/ngtemplateoutlet-type-checking-5d2dcb07a2c6
+sidebar:
+  order: 201
+---
+
+## Información
+
+:::note
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md:14]: same comment about `Note` translation

File: docs/src/content/docs/es/challenges/angular/6-permissions.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+Las Directivas Estructurales son in concepto importante que necesitaras dominar para mejorar tus habilidades y conocimientos en Angular. Esta es la primera parte del desafío.
+
+En Angular, "Guard" es un concepto muy importante por que siempre las vas a necesitar en cada aplicación que construyas.
+
+## Declaración
+
+En LoginComponent, encontraras 6 botones correspondientes a 6 tipos de roles de usuario:
+
+:::note
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/6-permissions.md:22]: Same as above

**Review by nelsongutidev — COMMENTED** [review]
Thanks for all your work on this!

File: docs/src/content/docs/es/challenges/angular/5-crud.md (hunk 8 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/5-crud.md:11]: Thanks Nelson, let me fix it.

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md (hunk 9 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:12]: do you have any doc site to confirm the syntax of this markup?

Comment (nelsongutidev) [reply docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:12]: Don't quite remember the docs reference but I implemented it here 

https://github.com/tomalaforge/angular-challenges/blob/main/docs/src/content/docs/es/challenges/performance/34-default-onpush.md?plain=1#L19

image

Comment (nelsongutidev) [reply docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md:12]: Found it :) https://starlight.astro.build/guides/authoring-content/#custom-aside-titles

File: docs/src/content/docs/es/challenges/angular/6-permissions.md (hunk 12 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/6-permissions.md]: Fixed

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md (hunk 11 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md]: Fixed

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md (hunk 10 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md]: Fixed

File: docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md (hunk 9 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/3-directive-enhancement.md]: Nice. Thanks!

Fixed.

File: docs/src/content/docs/es/challenges/angular/5-crud.md (hunk 8 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/5-crud.md]: Fixed

**Comment by ErickRodrCodes** [comment]
> Thanks for all your work on this!

Thanks, I think contributing to this project will widen up my current Angular knowledge.

**Review by nelsongutidev — APPROVED** [review]
Muchas gracias @ErickRodrCodes

**Comment by ErickRodrCodes** [comment]
> Muchas gracias @ErickRodrCodes
De nada! antes, muchas gracias por dejarme traducir estos desafios :)

De paso, por favor echale otra mirada y revisa si lo puedes aprovar. Gracias!

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+En este desafío, queremos aprender como reforzar el tipo de nuestro ng-template en AppComponent.
+
+Este desafío ofrece dos niveles de complejidad:
+
+### Nivel 1: una Interfase conocida
+
+A continuación tenemos estas lineas de código:
+
+!['Unkown' en 'Person'](../../../../assets/4/unknown-person.png 'Unkown' en 'Person')
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md:34]: Can we take a look at this? looks like the path needs updating since the `img` is not being shown...Its probably an extra `../` becuase of the es folder it lives on

image


english:
image

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+Como se puede observar, "name" es de tipo "any". La meta es inferir el tipo correcto.
+
+### Nivel 2: generic Interface
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md:38]: Can we update this title too?

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
+Actualmente, tenemos estas lineas de código:
+
+!['Unkown' en 'Student'](../../../../assets/4/unknown-student.png 'Unkown' en 'Student')
```
Comment (nelsongutidev) [inline docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md:42]: Looks like this path needs updating too, can we please take a look?

image

**Review by nelsongutidev — CHANGES_REQUESTED** [review]

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md (hunk 14 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md:38]: Como se me escapo eso lol

File: docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md (hunk 13 above)
Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md:34]: huh... when I run in Dev, even with the correct path, it is giving me the same kind of issue. I was able to fix it like this:

```md
![Any en Person](../../../../../assets/4/unknown-person.png)
```

but as a result, we won't have the tag title on the image. for some reason we can't push images with titles on it.

Comment (ErickRodrCodes) [reply docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md]: now it looks like this:
image
Probably we need to add a way to support a caption on the image?

**Review by nelsongutidev — APPROVED** [review]

**Comment by tomalaforge** [comment]
Thanks a lot for the amazing work you did 🙏

**Review by tomalaforge — APPROVED** [review]

**Comment by tomalaforge** [comment]
@all-contributors add @ErickRodrCodes for code

**Comment by ErickRodrCodes** [comment]
I MADE IT!
